SystemSettings.php

```diff
@@ -322,14 +342,21 @@ private function createBackendSetting()
         });
     }
 
-    private function createUseSentinelBackend()
-    {
-        return $this->makeSetting('useSentinelBackend', $default = false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
-            $field->title = Piwik::translate('QueuedTracking_UseSentinelFieldTitle');
-            $field->uiControl = FieldConfig::UI_CONTROL_CHECKBOX;
-            $field->uiControlAttributes = array('size' => 3);
+    private function createUseWhatRedisBackendType()
+    {        
+        $old_useSentinelBackend = new SystemSetting('useSentinelBackend', $default = false, FieldConfig::TYPE_BOOL, $this->pluginName);
+        $tmp_useWhatRedisBackendType = new SystemSetting('useWhatRedisBackendType', $default = 0, FieldConfig::TYPE_INT, $this->pluginName);
+        if ($tmp_useWhatRedisBackendType->getValue() == 0) {
+            $tmp_useWhatRedisBackendType->setValue($old_useSentinelBackend->getValue() == true ? 2 : 1);
+            parent::save();
+        }        
```

Comment:
This could be removed from this method and used in a migration script. Here's a somewhat similar example from another plugin: https://github.com/innocraft/plugin-MultiChannelConversionAttribution/blob/5.x-dev/Updates/4.2.1.php

Here's the developer documentation about creating a migration script: https://developer.matomo.org/guides/extending-database#defining-database-updates

Reply:
if we remove the code above and if someone just patches the code by copy & paste, without running the upgrade mechanism, it will also fail...

what do you think?

Reply:
That's good cautious thinking, but we mainly support upgrading via the application or extracting the release zip into the plugins directory. Although, manually patching should work as long as the plugin.json file is updated. When the version number in that file changes, the application checks for migration scripts and prompts to run them.

Reply:
i'm a little stuck here, the file name of this updater depends on the version number of the next release and i don't know what version this PR will be released in.

and i can't figure out how to run the updater script manually (just to confirm that it doesn't contain any errors)
for example, let's say i've created a file `5.0.8.php` under ["Updates"](https://github.com/haristku/plugin-QueuedTracking/tree/feature/add_redis_cluster_capability/Updates) folder, then how to run/
simulate it manually? what if i need to debug it first? i tried the `php console core:update` command with no luck...

```php
<?php
/**
 * Matomo - free/libre analytics platform
 *
 * @link https://matomo.org
 * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
 *
 */

namespace Piwik\Plugins\QueuedTracking;

use Piwik\Updater;
use Piwik\Updates as PiwikUpdates;
use Piwik\Updater\Migration\Factory as MigrationFactory;
use Piwik\Settings\Plugin\SystemSetting;
use Piwik\Settings\FieldConfig;

/**
 * Update for version 5.0.8.
 */
class Updates_5_0_8 extends PiwikUpdates
{
    public function __construct(MigrationFactory $factory) {}

    public function doUpdate(Updater $updater)
    {
        $old_useSentinelBackend = new SystemSetting('useSentinelBackend', $default = false, FieldConfig::TYPE_BOOL, 'QueuedTracking');
        $tmp_useWhatRedisBackendType = new SystemSetting('useWhatRedisBackendType', $default = 0, FieldConfig::TYPE_INT, 'QueuedTracking');
        if ($tmp_useWhatRedisBackendType->getValue() == 0) {
            $tmp_useWhatRedisBackendType->setValue($old_useSentinelBackend->getValue() == true ? 2 : 1);
            $tmp_useWhatRedisBackendType->save();
        }          
    }
}
```
or maybe i'll just leave this matter to you?
i need your help ✨🙇 ✨ 

@AltamashShaikh @snake14

Reply:
finally i was able to make the updater run as it should, and i managed to debug the updater process.
i confirmed that the updater script can migrate the Redis Sentinel configuration.
i'll leave the version numbering up to you.

ty.